File: exercism/clojure/perfect-numbers/src/perfect_numbers.clj
(ns perfect-numbers)

(defn classify [n] ;; <- arglist goes here
  (if (neg? n)
    (throw (IllegalArgumentException.))
    (let [somme (reduce + (filter (fn [d] (zero? (mod n d))) (range 1 n)))]
      (cond
        (= somme n) :perfect
        (> somme n) :abundant
        (< somme n) :deficient))))

File: exercism/fortran/perfect-numbers/perfect_numbers.f90

module perfect_numbers
  implicit none

contains

  integer function divisors_sum(n)
    integer :: i, n
    divisors_sum = 0
    do i = 1, n / 2
      if(mod(n, i) == 0) then
        divisors_sum = divisors_sum + i
      end if
    end do
  end function

  character(len=9) function classify(num)
    integer, intent(in) :: num
    integer :: s
    

    if ( num <= 0) then
      classify = "ERROR"
      return
    end if

    s = divisors_sum(num)
    if (s == num) then
      classify = "perfect"
    else if (s < num) then
      classify = "deficient"
    else
      classify = "abundant" 
    end if
  end function

end module
